File: frontend/src/api/nexus.ts
import { api } from "./client";

export interface AssistantTurn {
  role: "user" | "nexus";
  text: string;
}

export interface NexusChatRequest {
  query: string;
  history: AssistantTurn[];
}

export interface NexusChatResponse {
  reply: string;
  to?: string | null;
  toLabel?: string | null;
}

export async function askNexus(req: NexusChatRequest): Promise<NexusChatResponse> {
  const { data } = await api.post<NexusChatResponse>("/nexus/chat", req);
  return data;
}

File: backend/app/schemas/nexus.py
from pydantic import BaseModel, Field


class AssistantTurn(BaseModel):
    role: str = Field(description="'user' or 'nexus'")
    text: str


class NexusChatRequest(BaseModel):
    query: str = Field(..., description="The user's latest message")
    history: list[AssistantTurn] = Field(default_factory=list, description="Previous chat turns for context")


class NexusChatResponse(BaseModel):
    reply: str = Field(..., description="Nexus's text reply to the user")
    to: str | None = Field(None, description="An optional URL path to route the user to (e.g. '/dashboard')")
    toLabel: str | None = Field(None, description="The label for the route link (e.g. 'Open Mission Control')")
